src/rdpc/index.ts

```diff
@@ -106,24 +124,47 @@ export const indexRdpcData = async ({
     analysesFetcher,
   });
 
+  /** ---------- transform data to DonorInfoMap --------------- */
```

Comment:
This section has a lot of duplicated code, even more when you include the multiple calls to `getAllMergedDonorWithSpecimens` above.

Is there a way to write this as a sinlge method and then call it multiple times (once per analysis type)?